notebooks/source/truncated_distributions.ipynb

```diff
@@ -0,0 +1,1614 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #44.        @partial(jax.vmap, in_axes=(None, 0))
</pre><p>It is not clear to me why <code>in_axes=(None, 0)</code>. I think you can write a while loop with signature (n, cdf)</p><pre class="ql-syntax" spellcheck="false">def cond_fn(val):
    n, cdf = val
    return jnp.any(cdf &lt; u)

def body_fn(val):
    n, cdf = val
    n_new = jnp.where(cdf &lt; u, n + 1, n)
    return n_new, self.cdf(n_new)

</pre><p>which works for general shapes of <code>u</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyro-ppl/numpyro/pull/1272/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>